### PR TITLE
Fix Target attribute output

### PR DESCRIPTION
**Pull request by innovate-invent**
The current output does not contain enough information to describe an alignment. The current output is also not GFF3 compliant. This commit is in reference to #2301

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by innovate-invent** [comment]
@abretaud Thanks for the help. Sorry I forgot to set this to WIP. I have some pending commits that remove the code you are editing.

**Comment by innovate-invent** [comment]
Testing the latest change exposes a bug in BioPy URI encoding '+' in the attributes when it shouldn't be.
I am going to abandon this tool for a simpler awk script that converts the backbone to GFF as I don't need scoring.

**Comment by abretaud** [comment]
Is there an issue for this encoding problem in https://github.com/biopython/biopython?

**Comment by innovate-invent** [comment]
Looks like the BCBio library has been depreciated.
https://github.com/chapmanb/bcbb/blob/master/gff/README.rst

**Comment by erasche** [comment]
@innovate-invent yeah it's deprecated the replacement for it has completely different semantics, and the current version of bcbio works sufficiently for most purposes. I tried to swap to gffutils at one point but had no success.

File: tools/progressivemauve/xmfa2gff3.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def convert_xmfa_to_gff3(xmfa_file, relative_to='1', sequences=None, window_size
                 type="match", strand=parent['strand'],
                 qualifiers={
                     "source": "progressiveMauve",
-                    "target": label_convert.get(other['id'], other['id']),
+                    "Target": " ".join(map(str, [label_convert.get(other['id'], other['id']), other['start'], other['end'], '+' if other['strand'] > 0 else '-'])),
```
Comment (erasche) [inline tools/progressivemauve/xmfa2gff3.py]: Would it make sense to add this to the `_id_tn_convert` function?

Comment (innovate-invent) [reply tools/progressivemauve/xmfa2gff3.py]: Do you mean _id_tn_dict?
I don't think that would be right.

Comment (erasche) [reply tools/progressivemauve/xmfa2gff3.py]: Yep that one, sorry, typo.

**Comment by innovate-invent** [comment]
That is a standing bug in the aforementioned BCBio gff library.

**Comment by hexylena** [comment]
Huh, that's unfortunate. Would ±1 work instead? Or we could consider to
fork and maintain it, it will be a dependency for a while due to sheer
amount of legacy code I have running on that lib. Do you have a link to the
issue? (On mobile or I'd look myself)

tir. 23. jun. 2020, 18:25 skrev Nolan Woods <notifications@github.com>:

**Comment by innovate-invent** [comment]
I believe this issue relates to the problem we are seeing: https://github.com/chapmanb/bcbb/issues/86

**Comment by hexylena** [comment]
@innovate-invent ok, I've added a small monkey patch for bcbio with the following logic:

- scope:
  - very limited scope of what we're patching, already abstracted into a function for us.
  - tiny function to override
  - most importantly: we have significant control over what data is in the key values, we know it will be safe overall, so the more advanced things the original function does, does not matter for us.
- bcbio is deprecated
  - I'm not interested in patching, other patches are upstreamed there but new releases are not forthcoming. the author has deprecated it, we should respect their wish to not be responsible for it, and forking is a significant undertaking.
  - I've had another look at the recommended replacement and it looks a lot more usable than it used to, so, switching to it is now a definite possibility
  - but dozens of tools depend on it, so, I'll do the migration to the new library in one go.

This patch has languished (my fault, I'm sorry) and it should definitely get in :) With this patch, I believe the tests will pass again.

**Pull request merged**